File: organize_structure.js
const fs = require('fs');
const path = require('path');

const projectDir = path.join(__dirname);
process.chdir(projectDir);

console.log('=' .repeat(70));
console.log('NEXT.JS PROJECT STRUCTURE ORGANIZATION');
console.log('=' .repeat(70));
console.log(`Project Directory: ${projectDir}\n`);

// Step 1: Create directories
const dirsToCreate = [
    path.join(projectDir, 'app', 'lib'),
    path.join(projectDir, 'app', 'api', 'newsletters', '[id]'),
    path.join(projectDir, 'app', 'write'),
    path.join(projectDir, 'app', 'newsletters', '[id]'),
    path.join(projectDir, 'app', 'components'),
];

console.log('1. Creating directories...');
dirsToCreate.forEach(dir => {
    try {
        if (!fs.existsSync(dir)) {
            fs.mkdirSync(dir, { recursive: true });
            const relPath = path.relative(projectDir, dir);
            console.log(`   ✓ Created: ${relPath}`);
        } else {
            const relPath = path.relative(projectDir, dir);
            console.log(`   ✓ Already exists: ${relPath}`);
        }
    } catch (err) {
        console.error(`   ✗ Error creating ${dir}: ${err.message}`);
    }
});

// Step 2: Copy files
const fileMappings = [
    { src: 'lib.db.ts', dst: path.join('app', 'lib', 'db.ts') },
    { src: 'api.newsletters.route.ts', dst: path.join('app', 'api', 'newsletters', 'route.ts') },
    { src: 'api.newsletters.[id].route.ts', dst: path.join('app', 'api', 'newsletters', '[id]', 'route.ts') },
    { src: 'app.write.page.tsx', dst: path.join('app', 'write', 'page.tsx') },
    { src: 'app.newsletter.[id].page.tsx', dst: path.join('app', 'newsletters', '[id]', 'page.tsx') },
    { src: 'components.NewsletterCard.tsx', dst: path.join('app', 'components', 'NewsletterCard.tsx') },
    { src: 'components.PasswordModal.tsx', dst: path.join('app', 'components', 'PasswordModal.tsx') },
];

console.log('\n2. Copying files to new locations...');
fileMappings.forEach(({ src, dst }) => {
    try {
        const srcPath = path.join(projectDir, src);
        const dstPath = path.join(projectDir, dst);
        
        if (fs.existsSync(srcPath)) {
            fs.copyFileSync(srcPath, dstPath);
            console.log(`   ✓ Copied: ${src} → ${dst}`);
        } else {
            console.log(`   ✗ Source not found: ${src}`);
        }
    } catch (err) {
        console.error(`   ✗ Error copying ${src}: ${err.message}`);
    }
});

// Step 3: Print directory structure
console.log('\n3. Final Directory Structure (app/):\n');
console.log('─'.repeat(70));

function printTree(dirPath, prefix = '', ignoreList = new Set(['.git', 'node_modules', 'UI', '.next'])) {
    try {
        const entries = fs.readdirSync(dirPath)
            .filter(e => !e.startsWith('.') && !ignoreList.has(e))
            .sort();
        
        entries.forEach((entry, index) => {
            try {
                const fullPath = path.join(dirPath, entry);
                const isLast = index === entries.length - 1;
                const connector = isLast ? '└── ' : '├── ';
                console.log(`${prefix}${connector}${entry}`);
                
                const stat = fs.statSync(fullPath);
                if (stat.isDirectory()) {
                    const nextPrefix = prefix + (isLast ? '    ' : '│   ');
                    printTree(fullPath, nextPrefix, ignoreList);
                }
            } catch (err) {
                console.error(`Error processing ${entry}: ${err.message}`);
            }
        });
    } catch (err) {
        console.error(`Error reading directory ${dirPath}: ${err.message}`);
    }
}

const appPath = path.join(projectDir, 'app');
if (fs.existsSync(appPath)) {
    console.log('app/');
    printTree(appPath);
    console.log('─'.repeat(70));
} else {
    console.error('Error: app directory not found');
}

console.log('\n' + '=' .repeat(70));
console.log('✓ ORGANIZATION COMPLETE!');
console.log('=' .repeat(70));
console.log('\nDirectory structure has been successfully organized.');
console.log('All files have been copied to their new locations in app/');

File: organize_structure.py
#!/usr/bin/env python3
import os
import shutil
from pathlib import Path

project_dir = r"c:\Users\seank\OneDrive\Desktop\linkus_newletter"
os.chdir(project_dir)

print("=" * 70)
print("NEXT.JS PROJECT STRUCTURE ORGANIZATION")
print("=" * 70)

# Step 1: Create directories
dirs_to_create = [
    os.path.join(project_dir, "app", "lib"),
    os.path.join(project_dir, "app", "api", "newsletters", "[id]"),
    os.path.join(project_dir, "app", "write"),
    os.path.join(project_dir, "app", "newsletters", "[id]"),
    os.path.join(project_dir, "app", "components"),
]

print("\n1. Creating directories...")
for dir_path in dirs_to_create:
    try:
        os.makedirs(dir_path, exist_ok=True)
        print(f"   ✓ Created: {dir_path.replace(project_dir + chr(92), '')}")
    except Exception as e:
        print(f"   ✗ Error creating {dir_path}: {e}")

# Step 2: Copy files
file_mappings = [
    (os.path.join(project_dir, "lib.db.ts"), os.path.join(project_dir, "app", "lib", "db.ts")),
    (os.path.join(project_dir, "api.newsletters.route.ts"), os.path.join(project_dir, "app", "api", "newsletters", "route.ts")),
    (os.path.join(project_dir, "api.newsletters.[id].route.ts"), os.path.join(project_dir, "app", "api", "newsletters", "[id]", "route.ts")),
    (os.path.join(project_dir, "app.write.page.tsx"), os.path.join(project_dir, "app", "write", "page.tsx")),
    (os.path.join(project_dir, "app.newsletter.[id].page.tsx"), os.path.join(project_dir, "app", "newsletters", "[id]", "page.tsx")),
    (os.path.join(project_dir, "components.NewsletterCard.tsx"), os.path.join(project_dir, "app", "components", "NewsletterCard.tsx")),
    (os.path.join(project_dir, "components.PasswordModal.tsx"), os.path.join(project_dir, "app", "components", "PasswordModal.tsx")),
]

print("\n2. Copying files to new locations...")
for source, dest in file_mappings:
    try:
        if os.path.exists(source):
            shutil.copy2(source, dest)
            src_name = source.replace(project_dir + chr(92), "")
            dst_name = dest.replace(project_dir + chr(92), "")
            print(f"   ✓ Copied: {src_name} → {dst_name}")
        else:
            print(f"   ✗ Source not found: {source.replace(project_dir + chr(92), '')}")
    except Exception as e:
        print(f"   ✗ Error copying {source}: {e}")

print("\n3. Final Directory Structure (app/):")
print("-" * 70)

def print_tree(path, prefix="", ignore_dirs={".git", "node_modules", "UI"}):
    try:
        entries = sorted(os.listdir(path))
    except PermissionError:
        return

    entries = [e for e in entries if not e.startswith('.') and e not in ignore_dirs]
    
    for i, entry in enumerate(entries):
        entry_path = os.path.join(path, entry)
        is_last = i == len(entries) - 1
        connector = "└── " if is_last else "├── "
        print(f"{prefix}{connector}{entry}")
        
        if os.path.isdir(entry_path):
            next_prefix = prefix + ("    " if is_last else "│   ")
            print_tree(entry_path, next_prefix, ignore_dirs)

app_path = os.path.join(project_dir, "app")
print(f"app{os.sep}")
print_tree(app_path, "")

print("\n" + "=" * 70)
print("✓ ORGANIZATION COMPLETE!")
print("=" * 70)
